### PR TITLE
Fix/poolarray

### LINKED ISSUE #124: PoolArray hash functions are recursive
```kotlin
override fun hashCode(): Int {
    return hashCode()
}
```
is a recursive loop and will crash if invoked.

### PR DESCRIPTION
**Pull request by CedNaru**
Fix #124  #123 

The missing functions were not part of the PoolArray class, I had to retrieve them from the VariantCall structure.
No other choice than to call the function with a StringName.

### CODE REVIEW

File: kt/godot-runtime/src/main/kotlin/godot/core/pool/PoolByteArray.kt [hunk 1]
```diff
@@ -8,12 +8,35 @@ import godot.util.VoidPtr
...
+         */
+        COMPRESSION_ZSTD,
+
+        /**
+         * Uses the gzip compression method.
+         */
+        COMPRESSION_GZIP
+    }
+
+
```
Comment (chippmann) [inline kt/godot-runtime/src/main/kotlin/godot/core/pool/PoolByteArray.kt:33]: Hmm isn't the idea that the compression modes in `File` class are used? At least thats to where the link [CompressionMode](https://docs.godotengine.org/en/stable/classes/class_file.html#enum-file-compressionmode) in https://docs.godotengine.org/en/stable/classes/class_poolbytearray.html#class-poolbytearray-method-compress is linking.

**Review by chippmann — CHANGES_REQUESTED** [review]
I'd also add some gut tests for it? I think the ascii functions are a good candidate for it

File: kt/godot-runtime/src/main/kotlin/godot/core/pool/PoolByteArray.kt (hunk 1 above)
Comment (CedNaru) [reply kt/godot-runtime/src/main/kotlin/godot/core/pool/PoolByteArray.kt:33]: Yeah, it is in File. 
But PoolByteArray is a different module than the GenAPI. I can't access it from there.

**Review by chippmann — APPROVED** [review]

File: kt/godot-runtime/src/main/kotlin/godot/core/pool/PoolByteArray.kt (hunk 1 above)
Comment (piiertho) [reply kt/godot-runtime/src/main/kotlin/godot/core/pool/PoolByteArray.kt:33]: So maybe a comment here would help ?

File: src/bridges/pool_byte_array_bridge.cpp [hunk 2]
```diff
@@ -104,6 +148,78 @@ void PoolByteArrayBridge::engine_call_appendArray(JNIEnv* p_raw_env, jobject p_i
...
+}
+
+void PoolByteArrayBridge::engine_call_decompress(JNIEnv* p_raw_env, jobject p_instance, jlong p_raw_ptr) {
+    jni::Env env{p_raw_env};
+    Variant args[2] = {};
+    TransferContext* transfer_context{GDKotlin::get_instance().transfer_context};
+    transfer_context->read_args(env, args);
+
+    Variant pool{*from_uint_to_ptr<PoolByteArray>(p_raw_ptr)};
+    static StringName func_name{_scs_create("decompress")};
```
Comment (piiertho) [inline src/bridges/pool_byte_array_bridge.cpp:173]: Not needed anymore ?

File: harness/tests/src/main/kotlin/godot/tests/Invocation.kt [hunk 3]
```diff
@@ -1,538 +1,547 @@
 package godot.tests
 
-import godot.ARVRServer
-import godot.Button
-import godot.NavigationMesh
-import godot.Object
-import godot.Spatial
+import godot.*
```
Comment (piiertho) [inline harness/tests/src/main/kotlin/godot/tests/Invocation.kt:3]: Godot replaced spaces

File: kt/godot-runtime/src/main/kotlin/godot/core/pool/PoolByteArray.kt [hunk 4]
```diff
@@ -8,12 +8,35 @@ import godot.util.VoidPtr
...
     //PROPERTIES
     val size: Int
         get() {
-		    Bridge.engine_call_size(_handle)
-			return TransferContext.readReturnValue(VariantType.JVM_INT) as Int
-		}
+            Bridge.engine_call_size(_handle)
```
[the remark this answers is not in the document]
Comment (CedNaru) [reply kt/godot-runtime/src/main/kotlin/godot/core/pool/PoolByteArray.kt:37]: It's just the result of auto reformat.

**Review by piiertho — APPROVED** [review]

**Review by raniejade — APPROVED** [review]

**Pull request merged**